File: packages/language-identification/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/languageidentification/classes/results/IdentifyLanguageResult.java
package io.capawesome.capacitorjs.plugins.mlkit.languageidentification.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.mlkit.languageidentification.interfaces.Result;

public class IdentifyLanguageResult implements Result {

    @NonNull
    private final String language;

    public IdentifyLanguageResult(@NonNull String language) {
        this.language = language;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("language", language);
        return result;
    }
}

File: packages/language-identification/ios/Plugin/Classes/Results/IdentifyLanguageResult.swift
import Foundation
import Capacitor

@objc public class IdentifyLanguageResult: NSObject, Result {
    let language: String

    init(language: String) {
        self.language = language
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["language"] = language
        return result as AnyObject
    }
}
